main_test.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"testing"
+
+	steps_test "github.com/ONSdigital/dp-dataset-api/features/steps"
+	featuretest "github.com/armakuni/dp-go-featuretest"
+	"github.com/cucumber/godog"
+	"github.com/cucumber/godog/colors"
+)
+
+var componentFlag = flag.Bool("component", false, "perform component tests")
+
+type FeatureTest struct {
+	MongoFeature *featuretest.MongoFeature
+}
+
+func (f *FeatureTest) InitializeScenario(ctx *godog.ScenarioContext) {
+	authorizationFeature := featuretest.NewAuthorizationFeature()
+	datasetFeature, err := steps_test.NewDatasetFeature(f.MongoFeature, authorizationFeature.FakeAuthService.ResolveURL(""))
+	if err != nil {
+		panic(err)
+	}
+
+	apiFeature := featuretest.NewAPIFeature(datasetFeature.InitialiseService)
+
+	ctx.BeforeScenario(func(*godog.Scenario) {
+		apiFeature.Reset()
+		datasetFeature.Reset()
+		f.MongoFeature.Reset()
+		authorizationFeature.Reset()
+	})
+
+	ctx.AfterScenario(func(*godog.Scenario, error) {
+		datasetFeature.Close()
+		authorizationFeature.Close()
+	})
+
+	datasetFeature.RegisterSteps(ctx)
+	apiFeature.RegisterSteps(ctx)
+	authorizationFeature.RegisterSteps(ctx)
+}
+
+func (f *FeatureTest) InitializeTestSuite(ctx *godog.TestSuiteContext) {
+	ctx.BeforeSuite(func() {
+		f.MongoFeature = featuretest.NewMongoFeature(featuretest.MongoOptions{Port: 27017, MongoVersion: "4.0.5", DatabaseName: "testing"})
```

Comment:
Any particular reason why Mongo version `4.0.5` used? The current stable release for mongo is `4.4.` Also, it is important that when initializing a test suite, the test suite uses the mongo version that is currently being used by the service. Initialising the mongo version of the test suite in this particular manner means that in the future, it is quite easy to forget to update the version of mongo in the suite when it comes to it. Using constants would be ideal here

Reply:
version set as constant in the top of the file